modules/deployment-model/src/test/java/org/mule/runtime/deployment/model/internal/tooling/ToolingPluginArtifactClassLoaderTestCase.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.deployment.model.internal.tooling;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+import static org.mule.runtime.module.artifact.classloader.ClassLoaderLookupStrategy.PARENT_FIRST;
+import org.mule.runtime.deployment.model.api.DeploymentException;
+import org.mule.runtime.deployment.model.api.plugin.ArtifactPluginDescriptor;
+import org.mule.runtime.module.artifact.classloader.ArtifactClassLoader;
+import org.mule.runtime.module.artifact.classloader.ArtifactClassLoaderFilter;
+import org.mule.runtime.module.artifact.classloader.ClassLoaderLookupPolicy;
+import org.mule.runtime.module.artifact.classloader.DisposableClassLoader;
+import org.mule.runtime.module.artifact.classloader.RegionClassLoader;
+import org.mule.runtime.module.artifact.classloader.TestArtifactClassLoader;
+import org.mule.runtime.module.artifact.descriptor.ArtifactDescriptor;
+import org.mule.tck.junit4.AbstractMuleTestCase;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class ToolingPluginArtifactClassLoaderTestCase extends AbstractMuleTestCase {
+
+  private static final String PLUGIN_NAME = "test-plugin";
+  private static final String TEST_REGION = "test-region";
+  private static final String REGION_NAME = "test-region-descriptor";
+
+  private RegionClassLoader regionClassLoader;
+  private ArtifactPluginDescriptor artifactPluginDescriptor;
+  private TestToolingPluginClassLoader pluginArtifactClassLoader;
+
+  @Before
+  public void createAppClassLoader() {
+    final ClassLoaderLookupPolicy classLoaderLookupPolicy = mock(ClassLoaderLookupPolicy.class);
+    //mandatory to find a resource releaser instance when doing the {@link org.mule.runtime.module.artifact.classloader.RegionClassLoader#dispose}
+    when(classLoaderLookupPolicy.getLookupStrategy(anyString())).thenReturn(PARENT_FIRST);
+
+    regionClassLoader =
+        new RegionClassLoader(TEST_REGION, new ArtifactDescriptor(REGION_NAME), null, classLoaderLookupPolicy);
+    /**
+     * we simulate loading an additional classloader as the {@link ToolingPluginClassLoaderBuilder#build()} will do when
+     * creating the fake {@link ArtifactDescriptor}.
+     */
+    regionClassLoader.addClassLoader(mock(ArtifactClassLoader.class), mock(ArtifactClassLoaderFilter.class));
+    artifactPluginDescriptor = new ArtifactPluginDescriptor(PLUGIN_NAME);
+    pluginArtifactClassLoader = spy(new TestToolingPluginClassLoader(artifactPluginDescriptor));
+  }
+
+  @Test(expected = DeploymentException.class)
```

Comment:
It may not be easy because, but can you see if there's another type of exception that can be thrown? Ideally the process of creating a Classloader for an artifact is independent from "deploying" so I would expect another exception type and not a DeploymentException.